stdlib/domain.mli

```diff
@@ -39,27 +39,25 @@ val self : unit -> id
 (** [self ()] is the identifier of the currently running domain *)
 
 val at_first_spawn : (unit -> unit) -> unit
-(** Register the given function to be called before any domain is
-    spawned (except the initial domain) and before any function registered
-    with {!val:at_startup} is called. The functions registered with
-    [at_first_spawn] are called in 'last in, first out' order: the function
-    most recently added with [at_first_spawn] is called first.
+(** [at_first_spawn f] calls [f] before the first domain is spawned by the
+    program. The functions registed with [at_first_spawn] are called on the
+    main (initial) domain. The functions registered with [at_first_spawn] are
+    called in 'first in, first out' order: the oldest function added with
+    [at_first_spawn] is called first.
 
-    @raise Invalid_argument if the first domain has already been spawned. *)
+    @raise Invalid_argument if the program has already spawned a domain. *)
+
+val at_each_spawn : (unit -> unit) -> unit
+(** [at_each_spawn f] calls [f] on the newly spawned domains. The function [f]
+    is executed before the callback [g] specified in [spawn g] is executed. The
+    registered functions are called in 'first in, first out' order: the oldest
+    function added with [at_each_spawn] is called first. *)
 
 val at_exit : (unit -> unit) -> unit
-(** Register the given function to be called at when a spawned domain exits.
-    This function is also registered with {!Stdlib.at_exit}. If the registered
-    function raises an exception, the exceptions are ignored. The registered
-    functions are called in 'last in, first out' order: the function most
-    recently added with [at_exit] is called first. *)
-
-val at_startup : (unit -> unit) -> unit
-(** Register the given function to be called when a domain starts. This
-    function is called before the callback specified to [spawn f] is
-    executed. The registered functions are called in 'last in, first out'
-    order: the function most recently added with [at_startup] is
-    called first. *)
+(** [at_exit f] calls [f] when the current domain exits. Note that [at_exit]
+    callbacks are domain-local and only apply to the calling domain. The
+    registered functions are called in 'last in, first out' order: the function
+    most recently added with [at_exit] is called first. *)
```

Comment:
Maybe we could comment that it is possible to combine `at_each_spawn` and `at_exit` to register a cleanup function for each future domain.

Reply:
Done. Pushed a small example to show how to use `at_each_spawn` and `at_exit` together.